src/main/java/com/lalabrand/ecommerce/security/password_reset/PasswordResetService.java

```diff
@@ -0,0 +1,117 @@
+package com.lalabrand.ecommerce.security.password_reset;
+
+import com.lalabrand.ecommerce.security.refresh_token.RefreshTokenService;
+import com.lalabrand.ecommerce.user.User;
+import com.lalabrand.ecommerce.user.UserRequest;
+import com.lalabrand.ecommerce.user.UserService;
+import com.lalabrand.ecommerce.utils.EmailSenderService;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.transaction.Transactional;
+import org.apache.coyote.BadRequestException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.nio.file.AccessDeniedException;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.Random;
+
+@Service
+public class PasswordResetService {
+    private final Logger logger = LoggerFactory.getLogger(PasswordResetService.class);
+    private final PasswordResetTokenRepository passwordResetTokenRepository;
+    private final UserService userService;
+    private final PasswordEncoder passwordEncoder;
+    private final EmailSenderService emailSenderService;
+    private final RefreshTokenService refreshTokenService;
+    @Value("${password.reset.token.symbols}")
+    private String passwordResetTokenSymbols;
+    @Value("${reset.password.token.expiration.seconds}")
+    private Integer resetPasswordExpiration;
+
+
+    public PasswordResetService(PasswordResetTokenRepository passwordResetTokenRepository,
+                                UserService userService, PasswordEncoder passwordEncoder,
+                                EmailSenderService emailSenderService, RefreshTokenService refreshTokenService) {
+        this.passwordResetTokenRepository = passwordResetTokenRepository;
+        this.userService = userService;
+        this.passwordEncoder = passwordEncoder;
+        this.emailSenderService = emailSenderService;
+        this.refreshTokenService = refreshTokenService;
+    }
+
+    @Transactional
+    public boolean resetPasswordForUser(PasswordResetRequest passwordResetInput) throws AccessDeniedException, BadRequestException {
+        Optional<User> user = userService.findByEmail(passwordResetInput.getEmail());
+        if (user.isPresent()) {
+            Optional<PasswordResetToken> resetToken = passwordResetTokenRepository
+                    .findByTokenAndUser(passwordResetInput.getToken(), user.get());
+            if (resetToken.isPresent()) {
```

Comment:
I already mentioned that you have to wrap unintentional logic in `if`. In that case, it would be better to handle logic when `resetToken.isPresent()` is `false` under the `if` statement.